File: 3612-adjacent-increasing-subarrays-detection-i/adjacent-increasing-subarrays-detection-i.java
class Solution {
    public boolean hasIncreasingSubarrays(List<Integer> nums, int k) {
        int n = nums.size();

        for (int a = 0; a + 2 * k - 1 < n; a++)
            if (isStrictlyIncreasing(nums, a, a + k - 1) &&
                    isStrictlyIncreasing(nums, a + k, a + 2 * k - 1)) 
                return true;

        return false;
    }

    boolean isStrictlyIncreasing(List<Integer> nums, int l, int r) {
        for (int i = l + 1; i <= r; i++) 
            if (nums.get(i) <= nums.get(i - 1)) 
                return false;

        return true;
    }
}

File: 3612-adjacent-increasing-subarrays-detection-i/adjacent-increasing-subarrays-detection-i.py
class Solution:
    def hasIncreasingSubarrays(self, nums: List[int], k: int) -> bool:
        a = 0
        b = k
        count = 1

        while b + 1 < len(nums):
            if count == k:
                return True

            if nums[a] < nums[a + 1] and nums[b] < nums[b + 1]:
                count +=1
            else:
                count = 1

            a +=1
            b +=1

        if count == k:
            return True

        return False